hack/build-go.sh

```diff
@@ -22,6 +22,7 @@ if [[ -z "$@" ]]; then
 fi
 
 os::build::build_binaries "$@"
+OS_GOFLAGS="${OS_GOFLAGS:-} ${OS_IMAGE_COMPILE_GOFLAGS}" os::build::build_static_binaries "${OS_IMAGE_COMPILE_TARGETS[@]-}" "${OS_SCRATCH_IMAGE_COMPILE_TARGETS[@]-}"
```

Comment:
yeah, I spoke with Clayton about this. I'm going to make a different target called `build-platform` and a `build-platform.sh` ... sorting out some of the details now